src/components/Priority/Priority.js

```diff
@@ -1,116 +1,188 @@
-    padding: '10px 0',
-    margin: '10px 20px 10px 20px',
-    width: '150px',
+    padding: "10px 0",
+    margin: "10px 20px 10px 20px",
+    width: "150px"
   },
   label: {
-    textTransform: 'capitalize',
+    textTransform: "capitalize"
   },
   rail: {
     height: 4,
-    borderRadius: 10,
+    borderRadius: 10
   },
   mark: {
     height: 4,
     width: 4,
     borderRadius: 4,
-    margin: '0 0 0 -2px',
+    margin: "0 0 0 -2px"
   },
   markLabel: {
-    top: '-14px',
-    position: 'absolute',
-    fontSize: '12px',
+    top: "-14px",
+    position: "absolute",
+    fontSize: "12px",
     fontFamily: '"rennerbook", "Helvetica", "Arial", "sans-serif"',
-    lineHeight: '1',
+    lineHeight: "1"
   },
   offset: {
-    fontFamily: '"rennerbook", "Helvetica", "Arial", "sans-serif"',
+    fontFamily: '"rennerbook", "Helvetica", "Arial", "sans-serif"'
   },
   track: {
     height: 4,
-    borderRadius: 10,
-  },
-})
+    borderRadius: 10
+  }
+});
 
 const marksWithLabels = [
   {
     value: 0,
-    label: 'low',
+    label: "low"
   },
   {
     value: 25,
-    label: '',
+    label: ""
   },
   {
     value: 50,
-    label: 'medium',
+    label: "medium"
   },
   {
     value: 75,
-    label: '',
+    label: ""
   },
   {
     value: 100,
-    label: 'high',
-  },
-]
+    label: "high"
+  }
+];
 
 const marksWithoutLabels = [
   {
     value: 0,
-    label: '',
+    label: ""
   },
   {
     value: 25,
-    label: '',
+    label: ""
   },
   {
     value: 50,
-    label: '',
+    label: ""
   },
   {
     value: 75,
-    label: '',
+    label: ""
   },
   {
     value: 100,
-    label: '',
-  },
-]
+    label: ""
+  }
+];
 
 function valuetext(value) {
   return `${value}%`;
 }
-
-function Priority({ goalAddress, onClose }) {
-
-  const classes = useStyles()
+function Priority({
+  goalAddress,
+  onClose,
+  createGoalVote,
+  whoami,
+  fetchGoalVotes,
+  updateGoalVote
+}) {
+  const classes = useStyles();
 
   const priorityItemVars = [
-    { "priorityIcon": "urgency.svg", "priorityItemTitle": "Urgency" },
-    { "priorityIcon": "importance.svg", "priorityItemTitle": "Importance" },
-    { "priorityIcon": "impact.svg", "priorityItemTitle": "Impact" },
-    { "priorityIcon": "effort.png", "priorityItemTitle": "Effort" },
-  ]
-  const priorityItems = priorityItemVars.map((priorityItem, index) =>
-    <div className="priority_item">
-      <Icon className='priority_item_icon' name={priorityItem.priorityIcon} size='small' />
-      <div className="priority_item_title">{priorityItem.priorityItemTitle}</div>
+    { priorityIcon: "urgency.svg", priorityItemTitle: "Urgency" },
+    { priorityIcon: "importance.svg", priorityItemTitle: "Importance" },
+    { priorityIcon: "impact.svg", priorityItemTitle: "Impact" },
+    { priorityIcon: "effort.png", priorityItemTitle: "Effort" }
+  ];
+  const [value, setValue] = useState({
+    Urgency: 50,
+    Importance: 50,
+    Impact: 50,
+    Effort: 50
+  });
+  const hadleOnChange = e => {
```

Comment:
There is a MUCH easier way to get the value :)

```suggestion
  const handleOnChange = (e, value) => {
```

Reply:
https://material-ui.com/api/slider/

Reply:
You could use a technique called "closures" to handle the different cases/sliders. 

I'm thinking...
`const handleOnChange = (sliderType) => (e, value) => {
    ...
}

<Slider onChangeCommitted={handleOnChange("urgency")} />
<Slider onChangeCommitted={handleOnChange("impact")} />
...

```

Reply:
To replace `e.target.parentElement.parentElement.children[1].textContent`